ddl/column.go

```diff
@@ -450,6 +450,9 @@ func (d *ddl) doModifyColumn(t *meta.Meta, job *model.Job, col *model.ColumnInfo
 		return ver, infoschema.ErrColumnNotExists.GenByArgs(oldName, tblInfo.Name)
 	}
 
+	// We need the latest column's offset and state. This information can be obtained from the store.
```

Comment:
We can't set the offset in ddl.ModifyColumn, because the offset may change when we execute the ddl.doModifyColumn.